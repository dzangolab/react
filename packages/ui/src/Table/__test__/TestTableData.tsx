const data = [
  { id: 1, name: "jone" },
  { id: 2, name: "abhi" },
  { id: 3, name: "mike" },
  { id: 4, name: "tony" },
];

const biggerData = [
  { id: 1, name: "jone" },
  { id: 2, name: "abhi" },
  { id: 3, name: "mike" },
  { id: 4, name: "tony" },
  { id: 5, name: "tony" },
  { id: 6, name: "tony" },
  { id: 7, name: "tony" },
  { id: 8, name: "tony" },
  { id: 9, name: "tony" },
  { id: 10, name: "tony" },
  { id: 11, name: "tony" },
  { id: 12, name: "tony" },
  { id: 13, name: "tony" },
  { id: 14, name: "tony" },
  { id: 15, name: "tony" },
];

const fetcher = () => {
  return data;
};

const columns = [
  {
    accessorKey: "id",
    header: () => <span>ID</span>,
  },
  {
    accessorKey: "name",
    header: () => <span>Name</span>,
  },
];

export { columns, data, fetcher, biggerData };
