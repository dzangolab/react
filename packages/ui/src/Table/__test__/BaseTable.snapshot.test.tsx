import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import { biggerData, columns, fetcher } from "./TestTableData";
import { BaseTable, Table } from "../index";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(
    <Table
      columns={columns}
      data={biggerData}
      fetcher={fetcher}
      totalItems={biggerData.length}
      rowsPerPageOptions={[5, 10]}
    >
      <BaseTable />
    </Table>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
