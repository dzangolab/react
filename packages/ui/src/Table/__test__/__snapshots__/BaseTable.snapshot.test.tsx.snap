// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div
  className="table-wrapper sticky-header hide-scrollbar"
>
  <div>
    <div
      className="actions-header"
    />
    <table>
      <colgroup>
        <col
          width=""
        />
        <col
          width=""
        />
      </colgroup>
      <thead
        className=""
      >
        <tr>
          <th>
            <button
              onClick={[Function]}
            >
              <img
                src="/src/assets/images/adjustments.svg"
              />
            </button>
          </th>
        </tr>
        <tr
          className="row-header"
        >
          <th
            colSpan={1}
          >
            <div
              className=""
              onClick={[Function]}
            >
              <span>
                ID
              </span>
              <button>
                <img
                  src="/src/assets/images/arrow-up-down.svg"
                />
              </button>
              <div
                onClick={[Function]}
              >
                <button
                  onClick={[Function]}
                >
                  <img
                    src="/src/assets/images/filter.svg"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            colSpan={1}
          >
            <div
              className=""
              onClick={[Function]}
            >
              <span>
                Name
              </span>
              <button>
                <img
                  src="/src/assets/images/arrow-up-down.svg"
                />
              </button>
              <div
                onClick={[Function]}
              >
                <button
                  onClick={[Function]}
                >
                  <img
                    src="/src/assets/images/filter.svg"
                  />
                </button>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          data-id={1}
        >
          <td
            data-label="id"
          >
            1
          </td>
          <td
            data-label="name"
          >
            jone
          </td>
        </tr>
        <tr
          data-id={2}
        >
          <td
            data-label="id"
          >
            2
          </td>
          <td
            data-label="name"
          >
            abhi
          </td>
        </tr>
        <tr
          data-id={3}
        >
          <td
            data-label="id"
          >
            3
          </td>
          <td
            data-label="name"
          >
            mike
          </td>
        </tr>
        <tr
          data-id={4}
        >
          <td
            data-label="id"
          >
            4
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={5}
        >
          <td
            data-label="id"
          >
            5
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={6}
        >
          <td
            data-label="id"
          >
            6
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={7}
        >
          <td
            data-label="id"
          >
            7
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={8}
        >
          <td
            data-label="id"
          >
            8
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={9}
        >
          <td
            data-label="id"
          >
            9
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={10}
        >
          <td
            data-label="id"
          >
            10
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={11}
        >
          <td
            data-label="id"
          >
            11
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={12}
        >
          <td
            data-label="id"
          >
            12
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={13}
        >
          <td
            data-label="id"
          >
            13
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={14}
        >
          <td
            data-label="id"
          >
            14
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
        <tr
          data-id={15}
        >
          <td
            data-label="id"
          >
            15
          </td>
          <td
            data-label="name"
          >
            tony
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="table-footer-wrapper"
  >
    <div>
      <span>
        <strong>
          1 of 3 
        </strong>
      </span>
      <div>
        <span>
          Go to page:
        </span>
        <input
          disabled={false}
          max={3}
          min={1}
          onBlur={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value={1}
        />
        <select
          onChange={[Function]}
          value={5}
        >
          <option
            value={5}
          >
            Show
             
            5
          </option>
          <option
            value={10}
          >
            Show
             
            10
          </option>
        </select>
      </div>
    </div>
    <div>
      <button
        disabled={true}
        onClick={[Function]}
      >
        <img
          src="/src/assets/images/double-chevron-left.svg"
        />
      </button>
      <button
        disabled={true}
        onClick={[Function]}
      >
        <img
          src="/src/assets/images/chevron-left.svg"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <img
          src="/src/assets/images/chevron-right.svg"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <img
          src="/src/assets/images/double-chevron-right.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
